bank_statement_parse_camt/account_bank_statement_import.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+"""Add process_camt method to account.bank.statement.import."""
+##############################################################################
+#
+#    Copyright (C) 2013-2015 Therp BV <http://therp.nl>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import logging
+from openerp import models
+from .camt import CamtParser as Parser
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AccountBankStatementImport(models.TransientModel):
+    """Add process_camt method to account.bank.statement.import."""
+    _inherit = 'account.bank.statement.import'
+
+    def _parse_file(self, cr, uid, data_file, context=None):
+        """Parse a CAMT053 XML file."""
+        parser = Parser()
+        try:
+            _LOGGER.debug("Try parsing with camt.")
+            return parser.parse(data_file)
+        except ValueError:
+            # Not a camt file, returning super will call next candidate:
+            _LOGGER.debug("Statement file was not a camt file.")
```

Comment:
Accepted your suggestion on this point as well. A pity that odoo master is using exceptions to go through the different formats, although it is quite normal for a customer to have different formats for different banks, or even for different downloads for the same bank.